jax/scipy/sparse/test_gmres.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import jax
+import jax.numpy as jnp
+import pytest
+
+import linalg
+jax.config.update('jax_enable_x64', True)
+
+jax_dtypes = [jnp.float64, jnp.float32]#, np.float64]#, np.complex64, np.complex128]
+
+
+def test_gmres_on_small_known_problem(dtype):
+  """
+  GMRES produces the correct result on an analytically solved
+  linear system.
+  """
+  A = jax.numpy.array(([[1, 1], [3, -4]]), dtype=dtype)
+  b = jax.numpy.array([3, 2], dtype=dtype)
+  x0 = jax.numpy.ones(2, dtype=dtype)
+  n_kry = 2
+  maxiter = 1
+
+  @jax.tree_util.Partial
+  def A_mv(x):
+    return A @ x
+  tol = A.size*jax.numpy.finfo(dtype).eps
+  x, _ = linalg.gmres(A_mv, b, x0=x0, tol=tol, atol=tol, restart=n_kry,
+                      maxiter=maxiter)
+  solution = jax.numpy.array([2., 1.], dtype=dtype)
+  try:
+    np.testing.assert_allclose(x, solution, atol=tol)
+  except AssertionError as e:
+    print("Failed at dtype ", dtype)
+    print(e)
```

Comment:
Generally you should not be adding these assertion wrappers :)

Instead, use parameterized tests (see the existing test suite for examples).